### ISSUE TITLE
Parsing error on "ALTER TABLE ... ADD CONSTRAINT ... FOREIGN KEY ..."

### ISSUE THREAD

**Issue by davidfuhr** [sqlfluff/sqlfluff#3593]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

I got a parsing error with the following sql:

```sql
ALTER TABLE employee
    ADD CONSTRAINT fk_team
        FOREIGN KEY (team_id) REFERENCES team (id);
```

### Expected Behaviour

No parsing error.

### Observed Behaviour

Parsing error:

```
L:   6 | P:   9 |  PRS | Line 6, Position 9: Found unparsable section: 'FOREIGN
                       | KEY (team_id) REFERENCES team (i...'
```

### How to reproduce

If you take the following sql file:

```sql
CREATE TABLE team (id NUMBER(18) PRIMARY KEY, name VARCHAR2(200));
CREATE TABLE employee (id NUMBER(18) PRIMARY KEY, name VARCHAR2(200), team_id NUMBER(18));

ALTER TABLE employee
    ADD CONSTRAINT fk_team
        FOREIGN KEY (team_id) REFERENCES team (id);
```

and run sqlfluff you get a "parsing error"

```
❯ sqlfluff lint --dialect oracle alter-table-add-fk.sql
== [alter-table-add-fk.sql] FAIL                                                                                                                                                                                                             
L:   1 | P:  47 | L029 | Keywords should not be used as identifiers.
L:   2 | P:  51 | L029 | Keywords should not be used as identifiers.
L:   2 | P:  91 | L016 | Line is too long.
L:   5 | P:   5 | L003 | Expected 0 indentations, found 1 [compared to line 04]
L:   5 | P:   9 | L014 | Unquoted identifiers must be consistently lower case.
L:   5 | P:   9 | L029 | Keywords should not be used as identifiers.
L:   5 | P:  20 | L063 | Datatypes must be consistently upper case.
L:   6 | P:   9 |  PRS | Line 6, Position 9: Found unparsable section: 'FOREIGN
                       | KEY (team_id) REFERENCES team (i...'
L:   6 | P:  51 | L003 | Expected 0 indentations, found 2 [compared to line 04]
WARNING: Parsing errors found and dialect is set to 'oracle'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### Dialect

oracle

### Version

```bash
❯ python3 --version
Python 3.9.13ersion
❯ sqlfluff --version
sqlfluff, version 1.1.0
```

### Configuration

[sqlfluff]

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
